### ISSUE TITLE
Tracking based on Attentive Neural Processes

### ISSUE THREAD

**Issue by nih23** [nih23/deepFibreTracking#2]
@qrybe ANP seem to be a nice substitute for Kalman-like approaches and our LSTM. We'll look into these approaches as soon as your refactoring is done :) There's a PyTorch implementation of ANPs, too 👯 

> Neural processes (NPs) learn stochastic processes and predict the distribution of target output adaptively conditioned on a context set of observed input-output pairs. Furthermore, Attentive Neural Process (ANP) improved the prediction accuracy of NPs by incorporating attention mechanism among contexts and targets. In a number of real-world applications such as robotics, finance, speech, and biology, it is critical to learn the temporal order and recurrent structure from sequential data. However, the capability of NPs capturing these properties is limited due to its permutation invariance instinct. In this paper, we proposed the Recurrent Attentive Neural Process (RANP), or alternatively, Attentive Neural Process-RecurrentNeural Network(ANP-RNN), in which the ANP is incorporated into a recurrent neural network.

https://arxiv.org/abs/1910.09323
